### PR TITLE
Add setInputFiles to k6 browser module

### PR DESCRIPTION
**Pull request by ankur22**
- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `pnpm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

If adding a new definition:

- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

File: types/k6/experimental/browser.d.ts [hunk 1]
```diff
@@ -1131,6 +1148,37 @@ export interface ElementHandle extends JSHandle {
      */
     selectText(options?: ElementHandleOptions): void;
 
+    /**
+     * Sets the value of the file input to these files.
```
Comment (inancgumus) [inline types/k6/experimental/browser.d.ts:1152]:
```suggestion
     * Sets the file input element's value to the specified files.
```

File: types/k6/experimental/browser.d.ts [hunk 2]
```diff
@@ -1131,6 +1148,37 @@ export interface ElementHandle extends JSHandle {
...
+     *
+     * To work with local file on the file system, work with the experimental
+     * fs module to load and read the file's content.
```
Comment (inancgumus) [inline types/k6/experimental/browser.d.ts:1155]:
```suggestion
     * To work with local files on the file system, use the experimental
     * fs module to load and read the file contents.
```

File: types/k6/experimental/browser.d.ts [hunk 3]
```diff
@@ -1131,6 +1148,37 @@ export interface ElementHandle extends JSHandle {
...
+     *
+     * The elementHandle must an [input element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input).
```
Comment (inancgumus) [inline types/k6/experimental/browser.d.ts:1157]:
```suggestion
     * The {@link ElementHandle | element handle} must be an [input element](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input).
```

File: types/k6/experimental/browser.d.ts [hunk 4]
```diff
@@ -1131,6 +1148,37 @@ export interface ElementHandle extends JSHandle {
...
+     * @param files
+     * @param options
+     */
+    setInputFiles(files: File | File[], options?: {
+        /**
+         * Maximum time in milliseconds. Pass 0 to disable the timeout. Default
+         * is overridden by the setDefaultTimeout option on `BrowserContext` or
+         * `Page`. Defaults to 30000.
```
Comment (inancgumus) [inline types/k6/experimental/browser.d.ts:1165]: Should we link these, too? (`@link ...`)

File: types/k6/experimental/browser.d.ts [hunk 5]
```diff
@@ -1131,6 +1148,37 @@ export interface ElementHandle extends JSHandle {
...
+         */
+        timeout?: number;
+
+        /**
+         * Setting this to `true` will bypass the actionability checks (visible,
```
Comment (inancgumus) [inline types/k6/experimental/browser.d.ts:1170]: To be consistent with the `noWaitAfter`'s doc:

```suggestion
         * If set to `true`, it bypasses the actionability checks (visible,
```

File: types/k6/experimental/browser.d.ts [hunk 6]
```diff
@@ -1131,6 +1148,37 @@ export interface ElementHandle extends JSHandle {
...
+         * stable, enabled). Defaults to `false`.
+         */
+        force?: boolean;
+
+        /**
+         * If set to `true` and a navigation occurs from performing this action, it
+         * will not wait for it to complete. Defaults to `false`.
```
Comment (inancgumus) [inline types/k6/experimental/browser.d.ts:1177]: I believe present simple tense is more appropriate for docs as we don't know when the action will happen.

```suggestion
         * If set to `true` and a navigation occurs from performing this action, it
         * does not wait for it to complete. Defaults to `false`.
```

File: types/k6/experimental/browser.d.ts [hunk 7]
```diff
@@ -1479,6 +1527,40 @@ export interface Frame {
      */
     isVisible(selector: string, options?: StrictnessOptions): boolean;
 
+    /**
+     * Sets the value of the file input to these files.
+     *
+     * To work with local file on the file system, work with the experimental
+     * fs module to load and read the file's content.
```
Comment (inancgumus) [inline types/k6/experimental/browser.d.ts:1534]:
```suggestion
     * To work with local files on the file system, use the experimental
     * fs module to load and read the file contents.
```

File: types/k6/experimental/browser.d.ts [hunk 8]
```diff
@@ -1479,6 +1527,40 @@ export interface Frame {
      */
     isVisible(selector: string, options?: StrictnessOptions): boolean;
 
+    /**
+     * Sets the value of the file input to these files.
```
Comment (inancgumus) [inline types/k6/experimental/browser.d.ts:1531]:
```suggestion
     * Sets the file input element's value to the specified files.
```

File: types/k6/experimental/browser.d.ts [hunk 9]
```diff
@@ -1479,6 +1527,40 @@ export interface Frame {
...
+     *
+     * To work with local file on the file system, work with the experimental
+     * fs module to load and read the file's content.
+     *
+     * This method expects `selector` to point to an
```
Comment (inancgumus) [inline types/k6/experimental/browser.d.ts:1536]:
```suggestion
     * This method expects a `selector` to point to an
```

**Review by inancgumus — APPROVED** [review]
LGTM 👍

File: types/k6/experimental/browser.d.ts (hunk 5 above)
Comment (ankur22) [reply types/k6/experimental/browser.d.ts:1170]: I'm not sure about this option. Playwright has `timeout`, `noWaitAfter` and `strict` but not `force`. Maybe we should remove `force` from the type definitions and docs until we're sure that it's useful?

File: types/k6/experimental/browser.d.ts (hunk 4 above)
Comment (ankur22) [reply types/k6/experimental/browser.d.ts]: Added it in 6867bb061121b7ba780b49a15a2024903c331b62

File: types/k6/experimental/browser.d.ts (hunk 5 above)
Comment (inancgumus) [reply types/k6/experimental/browser.d.ts:1170]: I believe that might be a separate PR. WDYT?

Comment (ankur22) [reply types/k6/experimental/browser.d.ts:1170]: I'll remove `force` from this PR, but yeah, to add `strict` will need to be a new PR as we would need to implement the change in the k6 browser module first.

Comment (ankur22) [reply types/k6/experimental/browser.d.ts]: Done in 373707517c118664690b3f3f512d454dadcca040, 373707517c118664690b3f3f512d454dadcca040 and 4d0c05f5f714f4206f6fa41de560c0b76402b715

**Pull request merged**